src/koheesio/spark/transformations/debug.py

```diff
@@ -0,0 +1,118 @@
+    DataFrame data and schema:
+
+    Example
+    -------
+    ```python
+    from koheesio.spark.etl_task import EtlTask
+    from koheesio.spark.readers.dummy import DummyReader
+    from koheesio.spark.transformations.debug import Peek
+    from koheesio.spark.transformations.transform import Transform
+    from koheesio.spark.writers.dummy import DummyWriter
+
+    reader = DummyReader(n=10)
+    writer = DummyWriter()
+    transformations = [
+        Peek(enabled=True, enabled_by_logger=False, print_schema=True),
+        Transform(
+            lambda df: df.withColumn("id_squared", df["id"] * df["id"])
+        ),
+        Peek(enabled=True, enabled_by_logger=False, print_schema=True),
+    ]
+    task = EtlTask(
+        source=reader, transformations=transformations, target=writer
+    )
+    task.execute()
+    ```
+    """
+
+    enabled: bool = Field(
+        ...,
+        description="Whether to run this step or not. The resulting behavior may influenced by "
+        "the enabled_by_logger paramter and the current log level. See the class "
+        "docstring for more details.",
+    )
+
+    enabled_by_logger: Optional[bool] = Field(
+        default=False,
+        description="Enables the step if the log level of the logger is set to DEBUG or lower. The resulting behavior "
+        "may influenced by the enabled parameter and the current log level. See the class docstring for "
+        "more details.",
+    )
+
+    print_schema: Optional[bool] = Field(default=False, description="Enables printing the schema of the DataFrame")
+    truncate_output: Optional[bool] = Field(
+        default=False, description="Whether to truncate the output when printing the DataFrame"
+    )
+    vertical: Optional[bool] = Field(default=False, description="Print the DataFrame vertically")
+    n: Optional[int] = Field(default=20, description="Number of rows to show")
+
+    def execute(self) -> SparkStep.Output:
+        if self.active:
+            self._peek()
+
+    def _peek(self):
+        self.df = self.df.cache()
+
+        if self.print_schema:
+            self.df.printSchema()
+
+        self.df.show(truncate=self.truncate_output, n=self.n, vertical=self.vertical)
+        self.output.df = self.df
+
+    @property
+    def active(self) -> bool:
+        # The is active parameter takes precedence over the logger level
+        if not self.enabled:
+            return False
+
+        # Is active is True, and enabled by logger is False, we don't care about current log level
+        if not self.enabled_by_logger:
+            return True
+
+        # If both flags are True, it will be active based on log level
+        if self._on_debug():
+            return True
+
+        return False
+
+    def _on_debug(self) -> bool:
+        """
+        Check if the logger level is set to DEBUG or lower.
+        :return: True if the logger level is set to DEBUG or lower, False otherwise.
```

Comment:
please change this to Numpy notation

```markdown
Returns
--------
bool
    True if the logger level is set to DEBUG or lower, False otherwise.
```